docs/tutorials/agent_framework/RAG_with_conversational_flow_agent.md

```diff
@@ -0,0 +1,700 @@
+- So the population of New York City in 2023 (18,937,000) is much higher than the population of Seattle in 2023 (3,519,000)"""
+        }
+      ]
+    }
+  ]
+}
+```
+
+You can also customize which tool to use in predict API.
+For example, if you want to translate last answer into Chinese, you don't need to retrieve data from knowledge base.
+Then you can use `selected_tools` to specify just run `bedrock_claude_model`.
+
+Note: Agent will run tools sequentially with the new order defined in `selected_tools`. 
+
+```
+POST /_plugins/_ml/agents/your_agent_id/_execute
+{
+  "parameters": {
+    "question": "Translate last answer into Chinese?",
+    "selected_tools": ["bedrock_claude_model"]
+  }
+}
+```
+
+## 5. Advanced Topics
+### 5.1 Configure multiple knowledge bases
+You can configure multiple knowledge bases in agent. For example, if you have product description and comments data,
+you can configure like this 
+```
+{
+    "name": "My product agent",
+    "type": "conversational_flow",
+    "description": "This is an agent with product description and comments knowledge bases.",
+    "memory": {
+        "type": "conversation_index"
+    },
+    "app_type": "rag",
+    "tools": [
+        {
+            "type": "VectorDBTool",
+            "name": "product_description_vectordb",
+            "parameters": {
+                "model_id": "your_embedding_model_id",
+                "index": "product_description_data",
+                "embedding_field": "product_description_embedding",
+                "source_field": [
+                    "product_description"
+                ],
+                "input": "${parameters.question}"
+            }
+        },
+        {
+            "type": "VectorDBTool",
+            "name": "product_comments_vectordb",
+            "parameters": {
+                "model_id": "your_embedding_model_id",
+                "index": "product_comments_data",
+                "embedding_field": "product_comment_embedding",
+                "source_field": [
+                    "product_comment"
+                ],
+                "input": "${parameters.question}"
+            }
+        },
+        {
+            "type": "MLModelTool",
+            "description": "A general tool to answer any question",
+            "parameters": {
+                "model_id": "{{llm_model_id}}",
+                "prompt": "\n\nHuman:You are a professional product recommendation engine. You will always recommend product based on the given context. If you don't have enough context, you will ask Human to provide more information. If you don't see any related product to recommend, just say we don't have such product. \n\n Context:\n${parameters.product_description_vectordb.output}\n\n${parameters.product_comments_vectordb.output}\n\nHuman:${parameters.question}\n\nAssistant:"
+            }
+        }
+    ]
+}
+```
+When you run agent, it will query product description and comments data, then send query result and question to LLM.
```

Comment:
```suggestion
When you run the agent, it will query product description and comments data and then send query results and the question to the LLM.
```